app/models/concerns/profile_searching.rb

```diff
@@ -59,6 +59,24 @@ module ProfileSearching
     scope :os_major_version, lambda { |major, equals = true|
       where(benchmark: Xccdf::Benchmark.os_major_version(major, equals))
     }
+    scope :policy_bunch, lambda { |policy_or_profile_id|
+      return none unless UUID.validate(policy_or_profile_id)
```

Comment:
It's fine to validate here, but it'd be preferable to validate fields like this in the API layer (either GraphQL, REST, or both). REST uses https://github.com/zendesk/stronger_parameters to validate the shape of the request (i.e. `{ data: { attributes: { ... }}}`) and return some 4XX response.

Reply:
The validation there is to prevent database error, the same way as the builtin `where` does.
```
> Profile.where(id: 'bogus')
  Profile Load (1.9ms)  SELECT "profiles".* FROM "profiles" WHERE "profiles"."id" = $1  [["id", nil]]
=> []
```

There is a (thin) difference between a model relation scope and a controller. I agree on returning 4XX on incorrect values in controllers for most cases. Here we are focusing on internal and not much on the controller. The type checking should already have happen with the `ID` GraphQL type.